src/base_home.cpp

```diff
@@ -0,0 +1,136 @@
+#include "base_home.h"
+#include "submap.h"
+#include "mapdata.h"
+#include "map.h"
+
+#include <unordered_map>
+#include <list>
+#include <set>
+#include <queue>
+
+base_home::base(player &p, const tripoint &coreloc){
+    gen_overlay();
+    define_base_area(base_map, coreloc);
+    //TODO: Remove spawn points from base
+}
+
+/**
+ * Populates bmap.  (Add references to all the tiles in a submap)
+*/
+void gen_overlay(){
+    //TODO: Get help.
+}
+
+/**
+ * Populates bmap.bflag based on the rest of bmap.  Finds area of base
+ * 
+ * CAUTION: WILL RESET THE BASE!!! (mostly)
+ * TODO: Populate baflag with flags coresponding to the structure containing the command desk.
+ * TODO: Populate storage_open with locations of storage furnature (lockers etc.)
+ * TODO: Populate bunks with locations of sleeping areas (beds, cots, etc.)
+ */
+void base_home::define_base_area(const tripoint &coreloc) {
+    //First we map out the base floor by basically BFS all the "INDOORS" flags.
+    //Set up.
+    std::queue q;
+    std::set<tripoint> pushed; //not sure how big it needs to get before std::unordered_set is better than std::set.
+    tripoint cur, temp;
+    que.push(coreloc);
+    pushed.insert(coreloc);
+    //Make life easier for myself;
+    auto &x = cur.x;
+    auto &y = cur.y;
+    auto &z = cur.z;
+    auto &ter = bmap.ter;
+    auto &frn = bmap.fur;
+    auto &bf = bmap.bflag[x][y];
+    auto &in = base_flag::BASE_IN;
+    auto &wall = base_flag
```

Comment:
Don't use `auto` everywhere. It's useful when you have a giant structure, like `std::map<tripoint, std::string>::iterator`, but when it's just an `int`, it's much clearer when you just state it outright.